### PR TITLE
Solved minor issue and improved readme

### PR DESCRIPTION
**Pull request by eliax1996**

### CODE REVIEW

File: src/main/scala/com/github/tmnd1991/spark/testing/SparkFunSuite.scala [hunk 1]
```diff
@@ -17,19 +17,37 @@
...
-import org.scalatest.{ BeforeAndAfterAll, FunSuite, Outcome }
+import org.scalatest.{BeforeAndAfterAll, FunSuite, Outcome}
 
 /**
  * Base abstract class for all unit tests in Spark for handling common functionality.
  */
 abstract class SparkFunSuite extends FunSuite with BeforeAndAfterAll with Logging {
-
-  lazy val spark: SparkSession = SparkFunSuite.spark.newSession()
+  lazy val spark: SparkSession = {
```
Comment (tmnd1991) [inline src/main/scala/com/github/tmnd1991/spark/testing/SparkFunSuite.scala]: if sparkSession is a field of the SparkFunSuite, it will try to create a new instance for every test, but the new instance won't be created for every test (`getOrCreate` will `create` the first time and `get` all the subsequent calls), but it will register multiple shutdown hooks that will all try to stop the **same** spark session multiple times.
Also the fact that the call to `newSession` is skipped now, all the tests will share the same session, with also all the changes to configurations that might be made by other tests (i.e. having the side-effect that tests will interfere with each other more than it is now, causing some execution order related issues).

---

Bottom line: I agree on the new confs you are passing to start the test context, but I would like to keep the spark session inside the singleton (please also set spark.testing = true).

**Review by tmnd1991 — CHANGES_REQUESTED** [review]

**Comment by tmnd1991** [comment]
Thanks for the added docs, I see build is failing (duplicated test name) and I find the change on SparkSession initialisation not optimal.
Let me know if it is clear or discuss if you don't agree 😄

**Comment by eliax1996** [comment]
Yes I totally agree, I'll do it in the next days. 
I had stupidly copied the code that build in my machine without thinking of the state that a tests can inject inside the spark session.

**Comment by eliax1996** [comment]
Once I had 5 minutes in front of the screen I'll do it 😁
